src/frontend/DeepLink.ts

```diff
@@ -0,0 +1,52 @@
+//This needs to be opened on each initial screen (aka, when we have an
+//onboarding stack, this needs to be put on the initial screen of the onboarding
+//stack as well).
+import { Linking, Platform } from "react-native";
+import { NavigationNavigateAction, NavigationParams } from "react-navigation";
+import type {} from "react-navigation-stack";
+
+type nav = (
+  routeName:
+    | string
+    | {
+        action?: NavigationNavigateAction;
+        key?: string;
+        params?: NavigationParams;
+        routeName: string;
+      },
+  params?: NavigationParams,
+  action?: NavigationNavigateAction
+) => boolean;
+
+export function deepLinkSetUp(navigate: nav) {
+  console.log("This is being activated");
+  if (Platform.OS === "android") {
+    Linking.addEventListener("url", ({ url }) => {
+      handleUrl(navigate, url);
+    });
```

Comment:
Curious if the callback for add+remove event listener has to point to the same function object? I believe in the context of web, these two are not equivalent implementations:

```js
const handleUrl = () => {}

...
Foo.addEventListener('event', handleUrl)
...
Foo.removeEventLIstener('event', handleUrl())
```

vs.

```js
// This one is flawed because we're not technically removing the same handler in this case

Foo.addEventListener('event', () => handleUrl())
...
Foo.removeEventLIstener('event', () => handleUrl)
```